gpfit/tests/t_examples.py

```diff
@@ -48,9 +46,14 @@ def test_hoburgabbeel_ex6_1(self, example):
             setattr(TestExamples, name, old_test)  # move to a non-test fn
             delattr(TestExamples, fn)  # delete the old old_test
             new_name = "test_%s" % (name)
-            setattr(TestExamples, new_name, new_test(name, None))
+            setattr(TestExamples,
+                    new_name,
+                    logged_example_testcase(name=name,
+                                            imported=IMPORTED_EXAMPLES,
+                                            exampledir=EXAMPLE_DIR))
     TESTS.append(TestExamples)
```

Comment:
looks good. Only change I'd make is that GPfit probably doesn't need to run for all solvers -- just the default.

Do you want to go ahead and push your suggestion to the `testhelpers` branch over at GPkit?

Docstring for `generate_example_tests` could be
```
"""
Mutate TestCase class so it behaves as described in TestExamples docstring

Arguments
---------
path (str): directory containing example modules to test
testclass (class): a class that inherits from `unittest.TestCase`
newtest_fn (function): method that returns new tests, e.g. logged_example_testcase
solvers (iterable): solvers to run for; or only for default if solvers is None
"""
```